ros/lib/utils.py

```diff
@@ -151,18 +151,37 @@ def insert_performance_profiles(session, system_id, fields):
        performance_profile_history table.
     """
     fields = {} if fields is None else fields
+    fields_for_perf_profile_history = redact_instance_type_and_price(fields)
+
     old_profile_record = session.query(PerformanceProfile).filter_by(
         system_id=system_id).first()
     if old_profile_record:
         session.delete(old_profile_record)
         session.commit()
 
-    for model_class in [PerformanceProfile, PerformanceProfileHistory]:
-        new_entry = model_class(**fields)
+    model_fields_map = {PerformanceProfile: fields,
+                        PerformanceProfileHistory: fields_for_perf_profile_history}
+
+    for model_class, model_fields in model_fields_map.items():
+        new_entry = model_class(**model_fields)
         session.add(new_entry)
         session.flush()
 
 
+def redact_instance_type_and_price(fields):
+    """Remove unwanted attributes for performance profile history.
+        Such as - top_candidate & top_candidate_price.
+    """
+    fields_for_perf_profile_history = {}
+    try:
+        fields_for_perf_profile_history = fields.copy()
+        del fields_for_perf_profile_history['top_candidate']
+        del fields_for_perf_profile_history['top_candidate_price']
+    except KeyError:
+        LOG.debug("Key(s) top_candidate and/or top_candidate_price not found!")
```

Comment:
```suggestion
    except KeyError as err:
        LOG.debug(f'Key(s) not found in fields: {error}')
```

Can we get display key that actually not present using ^^ ?